### DISCUSSION TITLE
Change default Combobox behavior to not clear search term when focus moves

### DISCUSSION THREAD

**Discussion by dakahn** [carbon-design-system/carbon#7542] [Ideas]
Currently if the user enters a search term into the Carbon Combobox and moves focus to another element via keyboard or click the search term is cleared.

This behavior seems kinda unique. I couldn't find another Combobox in the wild that had similar functionality (including the WAI-Aria examples [seen here](https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html)). Youtube, Twitch, Github, and Twitter all do not clear the search term when focus is moved. 

So I'm proposing that we change this default behavior and allow the user to set the 'clear on focus move' prop if they want the old behavior.

**Comment by joshblack** [comment]
@dakahn is this a default that's coming through from Downshift or is it something our component is doing?

**Reply by dakahn** [reply]
Without looking at the code, it seems to be Downshift. 
https://codesandbox.io/s/6z67jvklw3

**Reply by joshblack** [reply]
Ah, that sucks. Definitely seems like something we should fix then so that the value is persistent 👍

**Comment by jnm2377** [comment]
I think this proposal makes sense. I would expect the search term to stay there after you move focus, unless it's been specifically cleared. Is it possible to override this if it's coming from Downshift?

**Reply by dakahn** [reply]
I'm not actually sure this isn't a Downshift bug actually (or a problem with our implementation). For a little background Downshift now supplies a special hook they expect combobox users to use called `useCombobox`. [It doesn't seem like there's a prop readily supplied to reverse this behavior](https://github.com/downshift-js/downshift/tree/master/src/hooks/useCombobox).

In fact in their documentation I don't really see anywhere they talk about this being purposeful. [They mimic the wai-aria docs](https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html) and mention replacing the text input with an empty string on `escape`
![image](https://user-images.githubusercontent.com/40970507/104381814-add26e00-54f2-11eb-94e0-8591dc16fa22.png)

But nothing about that happening on focus move.

**Comment by dakahn** [comment]
Worth noting is that we're discussing Combobox, but this behavior probably is in all our Downshift components (Multiselect etc)
